timesheet_automatic_reminder/models/timesheet_manager_reminder.py

```diff
@@ -0,0 +1,291 @@
+# -*- coding: utf-8 -*-
+# © 2015-2017 Elico corp (http://www.elico-corp.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import fields, models, api
+
+from common_method import *
+
+
+class TimesheetReport(models.TransientModel):
+    _name = "timesheet.manager.reminder.employee"
+    _description = "Timesheet Manager Reminder Employee list"
+
+    employee_id = fields.Many2one(
+        string='Employee',
+        comodel_name='hr.employee'
+    )
+
+    manager_id = fields.Many2one(
+        string='Manager',
+        comodel_name='timesheet.manager.reminder'
+    )
+
+    date_reminder = fields.Char(
+        string='Reminding date'
+    )
+
+
+class TimesheetReport(models.TransientModel):
+    _name = "timesheet.manager.reminder"
+    _description = "Timesheet Manager Reminder Service"
+    _inherit = ['mail.thread']
+
+    manager_id = fields.Many2one(
+        string='Manager',
+        comodel_name='hr.employee'
+    )
+
+    employee_list = fields.One2many(
+        'timesheet.manager.reminder.employee',
+        'manager_id',
+        string='Employee list'
+    )
+
+    created_date = fields.Char(
+        string='Create date'
+    )
+
+    @api.multi
+    def _check_weekend_or_holiday(
+            self,
+            selected_date=False,
+            employee_id=False
+    ):
+        publicholiday = self.env['hr.holidays.public']
+        if not publicholiday.is_public_holiday(
```

Comment:
@dongshukai Totally messy. Logic decisions, name of variables and returns should be understandable just on one view.

Absurd to read a public holiday could be or not a public holiday.